frontend/apps/filemanager/filemanagerhistory.lua

```diff
@@ -10,12 +10,30 @@ local KeyValuePage = require("ui/widget/keyvaluepage")
 local DocSettings = require("docsettings")
 local InfoMessage = require("ui/widget/infomessage")
 local T = require("ffi/util").template
-
-
+local RenderText = require("ui/rendertext")
+local Font = require("ui/font")
 local FileManagerHistory = InputContainer:extend{
     hist_menu_title = _("History"),
 }
 
+local ellipsis, space = "…", " "
+local ellipsis_width, space_width
+local function truncateTextByWidth(text, face, max_width, prepend_space)
```

Comment:
Looks like this logic will be needed in multiple widgets, let's move it into a method of rendertext module so it can be shared with filemanagerhistory and keyvaluepage ;)